Sources/CodableKit/Reflection/HiLoEncoder.swift

```diff
@@ -1,135 +1,136 @@
         }
-        
+
         mutating func nestedUnkeyedContainer(forKey key: Key) -> UnkeyedEncodingContainer {
             return UnkeyedEncoder(ctx, codingPath: codingPath + [key])
         }
-        
+
         mutating func superEncoder() -> Encoder {
-            return HiLoEncoder(ctx, codingPath: codingPath)
+            return HiLoEncoder<Root, Value>(ctx, codingPath: codingPath, keyPath: nil)
         }
-        
+
         mutating func superEncoder(forKey key: Key) -> Encoder {
-            return HiLoEncoder(ctx, codingPath: codingPath + [key])
+            return HiLoEncoder<Root, Value>(ctx, codingPath: codingPath + [key], keyPath: nil)
         }
     }
-    
+
     private struct UnkeyedEncoder: UnkeyedEncodingContainer {
         let codingPath: [CodingKey]
         let ctx: Context
         var count: Int
-        
+
         init(_ ctx: Context, codingPath: [CodingKey]) {
             self.ctx = ctx
             self.codingPath = codingPath
             self.count = 0
         }
-        
+
         mutating func encodeNil() throws {
             // ignore
         }
-        
+
         mutating func encode<T>(_ value: T) throws where T: Encodable {
             var c = SingleValueEncoder(ctx, codingPath: codingPath)
             try c.encode(value)
         }
-        
+
         mutating func nestedContainer<NestedKey>(keyedBy keyType: NestedKey.Type) -> KeyedEncodingContainer<NestedKey> where NestedKey : CodingKey {
             return .init(KeyedEncoder(ctx, codingPath: codingPath))
         }
-        
+
         mutating func nestedUnkeyedContainer() -> UnkeyedEncodingContainer {
             return UnkeyedEncoder(ctx, codingPath: codingPath)
         }
-        
+
         mutating func superEncoder() -> Encoder {
-            return HiLoEncoder(ctx, codingPath: codingPath)
+            return HiLoEncoder<Root, Value>(ctx, codingPath: codingPath, keyPath: nil)
         }
     }
-    
+
     private struct SingleValueEncoder: SingleValueEncodingContainer {
         var codingPath: [CodingKey]
         let ctx: Context
-        
+
         init(_ ctx: Context, codingPath: [CodingKey]) {
             self.ctx = ctx
             self.codingPath = codingPath
         }
-        
+
         mutating func encodeNil() throws {
             // ignore
         }
-        
+
         mutating func encode<T>(_ value: T) throws where T: Encodable {
-            if let custom = T.self as? AnyReflectionDecodable.Type {
+            if let custom = T.self as? AnyReflectionDecodable.Type,
+                custom.isBaseType || value is Value {
                 if !custom.anyReflectDecodedIsLeft(value) {
                     ctx.hi(codingPath)
                 }
             } else {
-                let encoder = HiLoEncoder(ctx, codingPath: codingPath)
+                let encoder = HiLoEncoder<Root, Value>(ctx, codingPath: codingPath, keyPath: nil)
```

Comment:
I think if you explicitly pass the generic arguments during init like this (`HiLoEncoder<Root, Value>`), you should not need to also pass the key path (which is never stored anyway).

If so, that could reduce the diff here a tiny bit more.

Reply:
You are correct